### PR TITLE
adds filter for showing only slow tests

### LINKED ISSUE #158: Add filter: hide regular passing tests, show slow passing tests
**Description**

My project has several thousand (unit and integration) tests and I really don't want to see a line for each passing test.
However, there are a few slow tests, and the "slowThreshold" feature is extremely helpful - thank you!

To be able to identify the slow tests, I'd like to see the corresponding output, while the regular (fast) tests are not displayed.

**Additional information**

Current:
```
SomeTest testA PASSED
SomeTest testB PASSED
SomeTest testC PASSED
SomeTest testD PASSED (502ms)
```

Expected:
```
SomeTest testD PASSED (502ms)
```

### PR DESCRIPTION
**Pull request by grimmjo**
closes #158

### CODE REVIEW

**Comment by grimmjo** [comment]
Hey @radarsh,

any news on this?
What needs to be done to get this PR merged?

Kind regards

**Comment by radarsh** [comment]
Hi @grimmjo, only feedback I have is to rename `showSlow` to `showOnlySlow` to be very clear. Please make these changes and it can then be merged. Thank you for the PR.

**Comment by grimmjo** [comment]
Hi @radarsh,
renaming is done and I also added the parameter to the README. I think we should be ready to merge 😃 
When do you plan to release the next version?

File: src/main/groovy/com/adarshr/gradle/testlogger/TestLoggerExtensionProperties.groovy [hunk 1]
```diff
@@ -84,4 +84,9 @@ abstract class TestLoggerExtensionProperties {
      * Whether simple class names should be used for displaying test suites. Defaults to false.
      */
     Boolean showSimpleNames
+
+    /**
+     * Whether only slow tests should be shown. Defaults to true.
```
Comment (radarsh) [inline src/main/groovy/com/adarshr/gradle/testlogger/TestLoggerExtensionProperties.groovy:89]: This should default to false. We're not showing _only_ slow tests all the time.

Comment (grimmjo) [reply src/main/groovy/com/adarshr/gradle/testlogger/TestLoggerExtensionProperties.groovy]: Oh my mistake. Default is set to false now.

**Comment by radarsh** [comment]
Hi @grimmjo there seems to be a problem. I'm running the tests with `showOnlySlow=true` and yet it's showing all tests, slow and fast.

```
➜  gradle-test-logger-plugin git:(slowFilter) ./gradlew clean test -Dtestlogger.showOnlySlow=true -Dtestlogger.slowThreshold=100

> Task :test

  com.adarshr.gradle.testlogger.renderer.AnsiTextRendererSpec

    render #input

      ✔ render [red]text (130ms)
      ✔ render [bright-red]text
      ✔ render [foo]text
      ✔ render [red,foo]text
      ✔ render [yellow]text
      ✔ render [bright-yellow]text
      ✔ render [green]text
      ✔ render [bright-green]text
      ✔ render [bold][red]text[/]
      ✔ render [bold,red]text[/]
      ✔ render [erase-all,bold,red]text[/]
      ✔ render [erase-ahead,bold,red]text[/]
      ✔ render [erase-back,bold,red]text[/]
      ✔ render [cursor-up-line]text[/]
      ✔ render [red][red][/]
      ✔ render [bold]text[bold-off]
      ✔ render [default]text
      ✔ render [grey]text
      ✔ render [cyan]text
      ✔ render [bright-cyan]text
      ✔ render [blue]text
      ✔ render [bright-blue]text
      ✔ render [magenta]text
      ✔ render [bright-magenta]text

  com.adarshr.gradle.testlogger.logger.OutputCollectorSpec

    ✔ output collector collects output by descriptor ID and removes them accordingly (912ms)

  com.adarshr.gradle.testlogger.logger.TestLoggerWrapperSpec

    ✔ wrapper delegates to sequential test logger if parallel theme is not applied (55ms)
    ✔ wrapper delegates to parallel test logger if parallel theme is applied

  com.adarshr.gradle.testlogger.logger.ConsoleLoggerSpec

    ✔ log
    ✔ does not log empty strings
    ✔ log new line
    ✔ configurable log level
```

File: README.md [hunk 2]
```diff
@@ -266,10 +267,11 @@ testlogger {
     showPassed false
     showSkipped false
     showFailed true
+    showOnlySlow false
 }
 ```
 
-By default all the above three flags are turned on. If you have chosen to display standard streams by setting
+By default all the above four flags are turned on. If you have chosen to display standard streams by setting
```
Comment (radarsh) [inline README.md:274]: `showOnlySlow` is will be off by default. So this goes back to being

> By default the flags `showPassed`, `showSkipped` and `showFailed` are turned on while `showOnlySlow` will be off.

File: src/main/groovy/com/adarshr/gradle/testlogger/TestResultWrapper.groovy [hunk 3]
```diff
@@ -21,7 +21,8 @@ class TestResultWrapper {
     boolean isLoggable() {
         testLoggerExtension.showPassed && testResult.successfulTestCount ||
             testLoggerExtension.showSkipped && (testResult.resultType == SKIPPED || testResult.skippedTestCount) ||
-            testLoggerExtension.showFailed && testResult.failedTestCount
+            testLoggerExtension.showFailed && testResult.failedTestCount ||
+            testLoggerExtension.showOnlySlow && (isTooSlow() || isMediumSlow())
```
Comment (radarsh) [inline src/main/groovy/com/adarshr/gradle/testlogger/TestResultWrapper.groovy:25]: There is something wrong with this expression.

**Review by radarsh — CHANGES_REQUESTED** [review]

**Comment by radarsh** [comment]
My understanding of this flag is:

Given `showOnlySlow=true` and `slowThreshold=5000`
And I have a test "test-1" that takes "10000" ms to run
And I have a test "test-2" that takes "3000" ms to run
When I run the tests
Then I should see "test-1" printed

and

Given `showOnlySlow=false` and `slowThreshold=5000`
And I have a test "test-1" that takes "10000" ms to run
And I have a test "test-2" that takes "3000" ms to run
When I run the tests
Then I should see "test-1" and "test-2" printed

File: src/main/groovy/com/adarshr/gradle/testlogger/TestLoggerExtension.groovy [hunk 4]
```diff
@@ -227,6 +229,15 @@ class TestLoggerExtension extends TestLoggerExtensionProperties {
             .getOrElse(false)
     }
 
+    Boolean getshowOnlySlow() {
```
Comment (radarsh) [inline src/main/groovy/com/adarshr/gradle/testlogger/TestLoggerExtension.groovy:232]: Typo in the method name :) This is why the command line argument wasn't working. It should be `getShowOnlySlow` (capitalised).

File: src/main/groovy/com/adarshr/gradle/testlogger/TestResultWrapper.groovy (hunk 3 above)
Comment (radarsh) [reply src/main/groovy/com/adarshr/gradle/testlogger/TestResultWrapper.groovy:25]: I was able to get it to work by changing the method body to be as below. If you can do it in a more readable manner, please do so.

```
          if (testLoggerExtension.showOnlySlow && (!isTooSlow() || !isMediumSlow())) {
              return false
          }
 
          testLoggerExtension.showPassed && testResult.successfulTestCount ||
              testLoggerExtension.showSkipped && (testResult.resultType == SKIPPED || testResult.skippedTestCount) ||
              testLoggerExtension.showFailed && testResult.failedTestCount
```

File: src/test/groovy/com/adarshr/gradle/testlogger/TestResultWrapperSpec.groovy [hunk 5]
```diff
@@ -127,6 +127,29 @@ class TestResultWrapperSpec extends Specification {
             10001         | false
     }
 
+    @Unroll
+    def "loggable returns false if showOnlySlow is turned #showOnlySlow"() {
```
Comment (radarsh) [inline src/test/groovy/com/adarshr/gradle/testlogger/TestResultWrapperSpec.groovy:131]: The tests will need fixing too

File: src/main/groovy/com/adarshr/gradle/testlogger/TestLoggerExtension.groovy (hunk 4 above)
Comment (grimmjo) [reply src/main/groovy/com/adarshr/gradle/testlogger/TestLoggerExtension.groovy:232]: My mistake. Done ;)

File: src/main/groovy/com/adarshr/gradle/testlogger/TestResultWrapper.groovy (hunk 3 above)
Comment (grimmjo) [reply src/main/groovy/com/adarshr/gradle/testlogger/TestResultWrapper.groovy:25]: Refactored this method a little bit to make it more readable

File: README.md (hunk 2 above)
Comment (grimmjo) [reply README.md:274]: done ;) just inserted your quote.

File: src/test/groovy/com/adarshr/gradle/testlogger/TestResultWrapperSpec.groovy (hunk 5 above)
Comment (grimmjo) [reply src/test/groovy/com/adarshr/gradle/testlogger/TestResultWrapperSpec.groovy:131]: done

**Comment by grimmjo** [comment]
My understanding of the flag is a little bit different as you may see in the `isLoggable` method.
First we need to check if either one of the flags `showPassed`, `showSkipped` or `showFailed` should be applied.
The `showOnlySlow` then filters the results from above.

What we should think about is, that we only should filter the passed test as the consumer may wants to see always the skipped and failed tests.

**Comment by radarsh** [comment]
You make a very good point here. 

> What we should think about is, that we only should filter the passed test as the consumer may wants to see always the skipped and failed tests.

We shouldn't hide failing tests because they failed fast :) But your PR seems to be already doing this.

**Review by radarsh — APPROVED** [review]

**Comment by grimmjo** [comment]
Hi @radarsh 😃 

belated happy new year ;) Do you have any plans on merging this pr and releasing a new version?

**Comment by radarsh** [comment]
@grimmjo Happy New Year. Yeah, let me get this done this week.

**Pull request merged**